aexpect/remote_door.py

```diff
@@ -0,0 +1,718 @@
+            break
+    if host_ip is None:
+        raise ValueError("No IP of the host machine was found")
+
+    logging.info("Sharing the test parameters over the network")
+    Pyro4.config.AUTOPROXY = False
+    Pyro4.config.REQUIRE_EXPOSE = False
+    try:
+        pyro_daemon = Pyro4.Daemon(host=host_ip, port=1437)
+        logging.debug("Pyro4 daemon started successfully")
+        uri = pyro_daemon.register(params)
+        pyrod_running = False
+    # address already in use OS error
+    except OSError:
+        pyro_daemon = Pyro4.Proxy("PYRO:" + Pyro4.constants.DAEMON_NAME +
+                                  "@" + host_ip + ":1437")
+        pyro_daemon.ping()
+        registered = pyro_daemon.registered()
+        logging.debug("Pyro4 daemon already started, available objects: %s",
+                      registered)
+        assert(len(registered) == 2)
+        assert(registered[0] == "Pyro.Daemon")
+        uri = "PYRO:" + registered[1] + "@" + host_ip + ":1437"
+        pyrod_running = True
+
+    if not pyrod_running:
+        logging.info("Starting the parameters provider for the remote host")
+        loop = DaemonLoop(pyro_daemon)
+        loop.start()
+
+    logging.debug("Pouring the params object to the host via uri %s", uri)
+    subcontrol = re.sub("URI[ \t\v]*=[ \t\v]*\".*\"", "URI = \"%s\"" % uri,
+                        subcontrol, count=1)
+
+    return subcontrol
+
+
+######################
+### REMOTE OBJECTS ###
+######################
+
+class DaemonLoop(threading.Thread):
+    """Loop thread for sharing remote objects."""
+
+    def __init__(self, pyro_daemon):
+        """
+        Contruct the Pyro daemon thread.
+
+        :param pyro_daemon: daemon for the remote python objects
+        :type pyro_daemon: Pyro4.Daemon object
+        """
+        threading.Thread.__init__(self)
+        self.pyro_daemon = pyro_daemon
+        self.daemon = True     # make this a Daemon Thread
+
+    def run(self):
+        """Run the Pyro4 daemon thread."""
+        self.pyro_daemon.requestLoop()
+
+
+def get_remote_object(object_name, session=None, host="localhost", port=9090):
+    """
+    Get a data object (visual or other) executing remotely or
+    share one if none is available, generating control files along
+    the way.
+
+    :param str object_name: name of the object to connect to
+    :param session: connection session if the object will be shared from the
+                    other side (devoted to communication with the object)
+    :type session: RemoteSession or None
+    :param str host: ip address of the local sharing server
+    :param int port: port of the local name server
+    :returns: proxy version of the remote object
+    :rtype: Pyro4.Proxy
+
+    To consider whitelisting of shareable functions/classes you have to share
+    local objects separately with more manual configuration on the sharing
+    process. This automatic two-way door opening considers simpler cases.
```

Comment:
Good opportunity to once again remind the user that this is a testing utility and should not be enabled in prod systems.

Reply:
Sure, I added this:
```diff
@@ -453,7 +458,8 @@ def get_remote_object(object_name, session=None, host="localhost", port=9090):
 
     To consider whitelisting of shareable functions/classes you have to share
     local objects separately with more manual configuration on the sharing
-    process. This automatic two-way door opening considers simpler cases.
+    process. This automatic two-way door opening considers simpler cases that
+    are focused on testing and not intended for use on production systems.
 
     This method does not rely on any static (template) controls in order to
     work because the remote door takes care to reach back the local one.
```